slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbConnection.java

```diff
@@ -61,12 +77,40 @@ public static void configureJdbi(final Jdbi jdbi) {
   }
 
   private static HikariDataSource createDataSource(
-      final String jdbcUrl, final String username, final String password) {
-    final HikariDataSource dataSource = new HikariDataSource();
-    dataSource.setJdbcUrl(jdbcUrl);
-    dataSource.setUsername(username);
-    dataSource.setPassword(password);
-    dataSource.addDataSourceProperty(PG_SOCKET_TIMEOUT_PARAM, DEFAULT_PG_SOCKET_TIMEOUT_SECONDS);
-    return dataSource;
+      final String jdbcUrl,
+      final String username,
+      final String password,
+      final Path hikariConfigurationFile) {
+    final Properties hikariConfigurationProperties =
+        loadHikariConfigurationProperties(hikariConfigurationFile);
+
+    final HikariConfig hikariConfig = new HikariConfig(hikariConfigurationProperties);
+    hikariConfig.setJdbcUrl(jdbcUrl);
```

Comment:
Looks like the config file can also set the jdbc url https://github.com/brettwooldridge/HikariCP#gear-configuration-knobs-baby. Which one will take precedence the command line or the property file?

Reply:
Added a test case that verifies that cli jdbc url (which is set using setJdbcUrl) will override value that might've been set by properties file.